### PR TITLE
Amoradzadeh/cp implicit filter

### PR DESCRIPTION
**Pull request by moradza**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

Reduce memory consumption of mixer kernels in context parallel setup.

**Collection**: [Note which collection this PR will affect]

# Changelog

- Add specific line by line info of high level changes in this PR.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: nemo/collections/llm/gpt/model/megatron/hyena/hyena_utils.py [hunk 1]
```diff
@@ -561,13 +561,22 @@ def get_t(self, L: int) -> torch.Tensor:
             self.t = t
             return self.t
 
-    def get_logp(self):
+    def get_logp(self, context_parallel_group=None):
         """Compute the log poles for the implicit modal filter."""
-        logp = -torch.exp(self.p.to(torch.float32))
-        glogp = logp * torch.exp(self.gamma.to(torch.float32))
+        if context_parallel_group is not None:
```
Comment (jstjohn) [inline nemo/collections/llm/gpt/model/megatron/hyena/hyena_utils.py]: should this be `and context_parallel_group.size() > 1:`?

Comment (moradza) [reply nemo/collections/llm/gpt/model/megatron/hyena/hyena_utils.py]: I see your point, if `_hyena_use_cp` set to False, this can be None. Slightly confusing code, b/c it sets cp_group to None.

**Review by jstjohn — DISMISSED** [review]

**Comment by moradza** [comment]
@jstjohn You can test locally CP == 1 and CP == 2 with this command. 
`torchrun --nproc_per_node=2 tests/collections/llm/gpt/model/test_hyena_mixer_cp.py --operator_type hyena [--use_subquadratic_ops]`

**Comment by pzelasko** [comment]
Closing this PR as "won't merge".
The following collections have been moved to separate repos in https://github.com/NVIDIA-NeMo organization: avlm, llm, multimodal, multimodal-autoregressive, vlm, speechlm, diffusion.
If you still wish to proceed with this contribution, please re-open it in the relevant repo.
